research/2d-decoding/rectangle_detector.py

```diff
@@ -157,33 +173,36 @@ def readImage(filename):
             continue
 
         try:
-            line = lambda a: (minX[1]-minY[1])/(minX[0]-minY[0])*(a - minX[0]) + minX[1] # Upper left side
+            def line(a): return (
+                minX[1]-minY[1])/(minX[0]-minY[0])*(a - minX[0]) + minX[1]  # Upper left side
             for x in range(minX[0], minY[0]+1):
                 y = round(line(x))
                 for offset in range(-LINE_PIXEL_TOLERANCE, LINE_PIXEL_TOLERANCE+1):
                     if y+offset < 0 or y+offset >= len(pixels):
                         continue
                     if pixels[y+offset][x] == 0:
                         break
-                else: # If no pixels within the tolerance are white, then the line isn't continuous, so this group isn't a rectangle
+                else:  # If no pixels within the tolerance are white, then the line isn't continuous, so this group isn't a rectangle
                     nottaSquare = True
                     break
         except ZeroDivisionError:
             continue
         if nottaSquare:
             continue
-        
+
         try:
             if minY[0]-maxX[0] != 0:
-                line = lambda a: (minY[1]-maxX[1])/(minY[0]-maxX[0])*(a - maxX[0]) + maxX[1] # Upper right side
+                # Upper right side
+                def line(a): return (minY[1]-maxX[1]) / \
+                    (minY[0]-maxX[0])*(a - maxX[0]) + maxX[1]
```

Comment:
At this point, why not make a function that handles all cases?

Reply:
Outdated + couldn't find the code in my codebase (I realize that these are all things fixed in the above layer, before I did the merge fix)